File: HorseRacing.Contracts/Models/Game/Dtos/GameEventDto.cs
using HorseRacing.Contracts.Base.Dto;

namespace HorseRacing.Contracts.Models.Game.Dtos
{
    public class GameEventDto : BaseDto
    {
        /// <summary>
        /// Порядковый номер шага (1,2,3,…)
        /// </summary>
        public int Step { get; private set; }
        /// <summary>
        /// Тип события
        /// </summary>
        public int EventType { get; private set; }
        /// <summary>
        /// Масть карты
        /// </summary>
        public int? CardSuit { get; private set; }
        /// <summary>
        /// Номинал карты
        /// </summary>
        public int? CardRank { get; private set; }
        /// <summary>
        /// Позиция карты в колоде
        /// </summary>
        public int? CardOrder { get; private set; }
        /// <summary>
        /// Масть лошади
        /// </summary>
        public int? HorseSuit { get; private set; }
        /// <summary>
        /// Позиция лошади
        /// </summary>
        public int? Position { get; private set; }
        /// <summary>
        /// Дата события
        /// </summary>
        public DateTime EventDate { get; private set; }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/dto/game-event-dto.ts
import type { BaseDto } from '~/interfaces/api/contracts/base/dto/base-dto';
import type { GameEventType } from '../enums/game-event-type-enum';
import type { SuitType } from '../enums/suit-type-enum';
import type { RankType } from '../enums/rank-type-enum';

export interface GameEventDto extends BaseDto {
    /** Порядковый номер шага (1,2,3,…) */
    Step: number;
    /** Тип события */
    EventType: GameEventType;
    /** Масть карты */
    CardSuit?: SuitType | null;
    /** Номинал карты */
    CardRank?: RankType | null;
    /** Позиция карты в колоде */
    CardOrder?: number | null;
    /** Масть лошади */
    HorseSuit?: SuitType | null;
    /** Позиция лошади */
    Position?: number | null;
    /** Дата события */
    EventDate: string;
}
